servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java

```diff
@@ -240,11 +249,56 @@ public StreamingHttpClient apply(final UrlKey urlKey) {
                 builder.sslConfig(DEFAULT_CLIENT_SSL_CONFIG);
             }
 
+            MultiAddressStrategyWrapper wrapper = new MultiAddressStrategyWrapper();
+            builder.appendClientFilter(wrapper);
+
             if (singleAddressInitializer != null) {
                 singleAddressInitializer.initialize(urlKey.scheme, urlKey.hostAndPort, builder);
             }
 
-            return builder.buildStreaming();
+            StreamingHttpClient singleClient = builder.buildStreaming();
+
+            wrapper.clientStrategy = singleClient.executionContext().executionStrategy();
+            return singleClient;
+        }
+    }
+
+    private static final class MultiAddressStrategyWrapper implements StreamingHttpClientFilterFactory {
+
+        HttpExecutionStrategy clientStrategy = defaultStrategy();
+        @Override
+        public StreamingHttpClientFilter create(final FilterableStreamingHttpClient client) {
+            return new StreamingHttpClientFilter(client) {
+                @Override
+                protected Single<StreamingHttpResponse> request(
+                        final StreamingHttpRequester delegate, final StreamingHttpRequest request) {
+                    applyMultiAddressStrategy(request.context(), clientStrategy);
+                    return super.request(delegate, request);
```

Comment:
when we implement a filter, the recommended approach is to use `delegate.request(request)` to avoid any unexpected behavior changes if the `super` implementation changes